src/memory/AmdFlash.hh

```diff
@@ -262,14 +278,18 @@ public:
 	};
 
 	struct Sector {
-		size_t address;
-		power_of_two<size_t> size;
-		bool writeProtect;
+		size_t address = 0;
+		power_of_two<size_t> size = 1;
+		bool writeProtect = false;
+		std::optional<bool> erasePending = {};
 		ptrdiff_t writeAddress = -1;
 		const uint8_t* readAddress = nullptr;
 
 		std::weak_ordering operator<=>(const Sector& sector) const { return address <=> sector.address; }
 		bool operator==(const Sector& sector) const { return address == sector.address; }
+
+		template<typename Archive>
+		void serialize(Archive& ar, unsigned version);
 	};
```

Comment:
Not sure if this is a good idea, but something to consider.

Most of this `struct Sector` is fixed (set once during the constructor and then never changes again).
In fact I think only the `erasePending` field still changes after the constructor, right?

Would it make sense to split this struct into a mutable and immutable part? And have two parallel `sectors` vectors (with two different names of course). That would also simplify the serialization code for (the mutable part of) `sectors`.

Reply:
Couple of thoughts:

1. If we have separate arrays, then we may have to rely more on indexing instead of those nice range-fors (guess we could use `zip`), and also do more checks to ensure they have the same size (e.g. post serialization).

2. It may be interesting to explore extracting an `EraseCommand { index; buffer { mark[] } }` and `ProgramCommand { index; address; buffer { opt<value>[] } }` class.

3. In the old [sparse persistence](https://github.com/openMSX/openMSX/pull/1629#discussion_r1609524490) pull request (that I do want to revisit sometime), I am also storing a mutable `modified` flag on sectors. Though if we would follow my idea in the second comment there, we would track address ranges separately.